### PR TITLE
Updates to Code generation, asyn/await

### PR DESCRIPTION
**Pull request by DServy**

### CODE REVIEW

**Comment by DServy** [comment]
@ChuckJonas Give'r a looksee

File: src/lib/sObject.ts [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
-export abstract class SObject{
-    constructor(type?: string) {
+export abstract class SObject {
+    constructor(type: string) {
 
         this.attributes = new SObjectAttributes();
         this.attributes.type = type;
     }
-    public Id?: string;
+    public id: string | undefined;
```
Comment (ChuckJonas) [inline src/lib/sObject.ts]: why `| undefined` over `?`

File: src/lib/sObject.ts [hunk 2]
```diff
@@ -23,44 +23,43 @@ export interface DMLResponse {
     success: boolean;
 }
 
-export class RestObject extends SObject{
+export class SObjectClient extends SObject {
```
Comment (ChuckJonas) [inline src/lib/sObject.ts]: this needs to go back to  `RestObject`

File: src/lib/sObject.ts [hunk 3]
```diff
@@ -23,44 +23,43 @@ export interface DMLResponse {
...
+        try {
+            const response = await this.generateCall(`/sobjects/${this.attributes.type}/${this.id}?_HttpMethod=PATCH`, data)
+            return response.data
+        } catch (error) {
+            console.log(error.response.data)
+            return error
+        }
+    }
+    public generateCall(path: string, data: SObject): Promise<AxiosResponse> {
+        return Rest.Instance.request.post(path, data);
```
Comment (ChuckJonas) [inline src/lib/sObject.ts]: should probably pass the method into this as well for things like `delete` & `refresh`

**Pull request merged**